src/runtime_ccall.cpp

```diff
@@ -146,21 +146,27 @@ void *jl_get_library(const char *f_lib)
 #endif
     if (f_lib == NULL)
         return jl_RTLD_DEFAULT_handle;
-    hnd = libMap[f_lib];
+    JL_LOCK_NOGC(&libmap_lock);
```

Comment:
this should probably be a `uv_mutex_t` or just a system mutex (`jl_thread_self` is not initialized on windows for threads not started by libuv), since this function could get called from a non-managed thread (e.g. `uv_async_send`)